zipline/utils/events.py

```diff
@@ -466,18 +466,15 @@ def should_trigger(self, dt, env):
             # of the simulation start
             self.calculate_start_and_end(dt, env)
 
-            # If we've missed the first trigger because it occurs before the
-            # simulation starts, recalculate for the next week
-            if dt > self.next_date_end:
-                self.calculate_start_and_end(dt + datetime.timedelta(days=7),
-                                             env)
-
-        # if the given dt is within the next matching day, return true. Also
-        # calculate the start and end dates for the next trigger
+        # If we've passed the trigger, calculate the next one
+        if dt > self.next_date_end:
```

Comment:
This is about guaranteeing that `calculate_start_and_end` returns a useful value, correct? I don't think it would be necessary. In any case, we still need the `if dt> self.next_date_end: ...` because that's what allows us to recalculate throughout the simulation. Adding this to `calculate_start_and_end` would be unnecessary, since we do it here (in `should_trigger`) anyway.

The way I see it, the responsibility of `calculate_start_and_end` is, given a date and an offset, to find the day of the week to set the trigger. It should not care whether or not this trigger has expired. `should_trigger`, on the other hand, should be responsible for recalculating the trigger once it has passed